crates/symbolicator/src/services/download/s3.rs

```diff
@@ -116,37 +120,45 @@ impl S3Downloader {
             );
             let s3 = Arc::new(match key.aws_credentials_provider {
                 AwsCredentialsProvider::Container => {
-                    let container_provider = rusoto_credential::ContainerProvider::new();
-                    let provider =
-                        match rusoto_credential::AutoRefreshingProvider::new(container_provider) {
-                            Ok(provider) => provider,
-                            Err(err) => panic!(
-                            "Unable to instantiate rusoto_credential::AutoRefreshingProvider: {:?}",
-                            err
-                        ),
-                        };
-                    self.create_s3_client(provider, region)
+                    let provider = LazyCachingCredentialsProvider::builder()
+                        .load(provide_credentials_fn(|| async {
+                            aws_config::ecs::EcsCredentialsProvider::builder()
```

Comment:
we should check with the folks who added the `AutoRefreshingProvider` and ask them to test this to see if it works equivalently.  I recall at some point they had an outage around this.  git blame should point in the right direction.

Reply:
I'll investigate this. Once I have this branch working well enough, I can also test this myself to some extent.

Reply:
This is still pending me looking into this.

Reply:
I am keeping an eye on the seemingly-related https://github.com/awslabs/aws-sdk-rust/issues/629.